salt/cloud/clouds/ec2.py

```diff
@@ -2784,7 +2782,7 @@ def destroy(name, call=None):
         )
 
     node_metadata = _get_node(name)
-    instance_id = node_metadata[name]['instanceId']
+    instance_id = node_metadata['instanceId']
```

Comment:
We probably need one style of code throughout the implementation. Not sure what to do here though.